library/modules/bigip_lx_package.py

```diff
@@ -346,18 +369,18 @@ def check_file_exists_on_device(self):
         resp = self.client.api.post(uri, json=params)
         try:
             response = resp.json()
-        except ValueError:
-            return False
-        if resp.status == 404 or 'code' in response and response['code'] == 404:
-            return False
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
 
-        try:
-            if "No such file or directory" in response['commandResult']:
+        if resp.status in [200, 201] or 'code' in response and response['code'] in [200, 201]:
+            try:
+                if "No such file or directory" in response['commandResult']:
+                    return False
+                if self.want.package_file in response['commandResult']:
+                    return True
+            except KeyError:
                 return False
```

Comment:
still not changes as per my last comment

Reply:
I don't know why that change didn't go through! A copy paste error, I guess. Fixed now!